### PR TITLE
chore: setup dependabot and GHA CI

### PR DESCRIPTION
**Pull request by bajtos**
Based on https://github.com/filecoin-station/spark-stats/tree/main/.github,
with few tweaks to try out a better setup.

Links:
- https://github.com/space-meridian/roadmap/issues/115

### CODE REVIEW

File: .github/workflows/dependabot-auto-approve-minor.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    strategy:
+      matrix:
+        dependencyStartsWith:
+          - '@sentry/'
+          - standard
```
Comment (bajtos) [inline .github/workflows/dependabot-auto-approve-minor.yml:15]: Added `standard` to the list of auto-approved minor upgrades.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  lint-all:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - run: npm ci
+      - run: npm run lint
```
Comment (bajtos) [inline .github/workflows/ci.yml:29]: Removed `npm run test:types` because we run tsc as part of the `lint` script.

**Comment by bajtos** [comment]
@juliangruber hindsight please 🙏🏻

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: CI
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```
Comment (juliangruber) [inline .github/workflows/ci.yml:6]: this limitation isn't necessary, as it prevents PRs targeting other branches from running CI

**Review by juliangruber — COMMENTED** [review]
Out of curiosity, why did you go for a multiple workspaces layout here?

**Comment by bajtos** [comment]
> Out of curiosity, why did you go for a multiple workspaces layout here?

I should have explained that in the design doc.

Cross-posting https://github.com/filecoin-station/piece-indexer/pull/3#issue-2504575385:

> Components:
> 
> * **indexer**: background service owning the persisted data, similar to spark-evaluate and spark-observer
> * REST API server exposing that data - to be done later

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bajtos) [reply .github/workflows/ci.yml:6]: Do you know why we have that limitation in spark-evaluate config?

https://github.com/filecoin-station/spark-evaluate/blob/f87560ec447d41ecaaadd3eafc7ab33d8eb2e078/.github/workflows/ci.yml#L5-L6

But not in spark-api config:

https://github.com/filecoin-station/spark-api/blob/0d79f8b060aac01d2f8c04d622aed3aded3dc707/.github/workflows/ci.yml#L5-L6

Git blame pointed me to this PR:

https://github.com/filecoin-station/spark-evaluate/pull/198

Comment (bajtos) [reply .github/workflows/ci.yml:6]: https://github.com/filecoin-station/piece-indexer/pull/8

**Pull request merged**